File: Parallel Programming/Lab06/NumInt/NumIntSeq.py
from sys import argv, exit
from time import perf_counter
from math import pi as mathPi


def main(argv):

    numberOfSteps = 10000

    '''
    #command line arguments
        
    if len(argv) != 2:
        print('Usage: {} <number of steps>' .format(argv[0]))
        exit(1)
    numberOfSteps = argv[1]
    try:
       numberOfSteps = int(numberOfSteps)
    except ValueError as e:
        print('Integer convertion error: {}' .format(e))
        exit(2)
    if numberOfSteps <= 0:
        print('Steps cannot be non-positive.')
        exit(3)
    '''
    
    t1 = perf_counter()

    pi = calcPi(numberOfSteps)

    t2 = perf_counter()

    print('Sequential program results with {} steps' .format(numberOfSteps))
    print('Computed pi = {}' .format(pi))
    print('Difference between estimated pi and math.pi = {}' .format(abs(pi - mathPi)))
    print('Time to compute = {} seconds' .format(t2-t1))


def calcPi(steps):

    sum = 0
    step = 1.0 / steps

    for i in range(steps):
        x = (i + 0.5) * step
        sum += 4.0 / (1.0 + x**2)

    pi = sum * step
    return pi


if __name__ == '__main__':
    main(argv)

File: Parallel-Programming/Lab06/NumInt/NumIntSeq.java
public class NumIntSeq {

    public static void main(String[] args) {

        long numSteps = 10000;
        double sum = 0.0;

        /* parse command line
        if (args.length != 1) {
		System.out.println("arguments:  number_of_steps");
                System.exit(1);
        }
        try {
		numSteps = Long.parseLong(args[0]);
        } catch (NumberFormatException e) {
		System.out.println("argument "+ args[0] +" must be long int");
		System.exit(1);
        }*/
        
        /* start timing */
        long startTime = System.currentTimeMillis();

        double step = 1.0 / (double)numSteps;
        /* do computation */
        for (long i=0; i < numSteps; ++i) {
            double x = ((double)i+0.5)*step;
            sum += 4.0/(1.0+x*x);
        }
        double pi = sum * step;

        /* end timing and print result */
        long endTime = System.currentTimeMillis();
        System.out.printf("sequential program results with %d steps\n", numSteps);
        System.out.printf("computed pi = %22.20f\n" , pi);
        System.out.printf("difference between estimated pi and Math.PI = %22.20f\n", Math.abs(pi - Math.PI));
        System.out.printf("time to compute = %f seconds\n", (double) (endTime - startTime) / 1000);
    }
}
